src/Field.js

```diff
@@ -100,6 +100,29 @@ export default class Field extends React.Component<Props, State> {
     },
     onChange: (event: SyntheticInputEvent<*> | any) => {
       const { parse, value: _value } = this.props
+
+      if (process.env.NODE_ENV !== 'production' && event && event.target) {
+        const targetType = event.target.type
+        const props: Object = this.props
+        const unknown =
+          ~['checkbox', 'radio', 'select-multiple'].indexOf(targetType) &&
+          !props.type
+
+        const type = targetType === 'select-multiple' ? 'select' : targetType
+        const { value }: any =
```

Comment:
aint sure if this is entirely correct, if not - let me know what should be changed about it